src/net/tcp/stream.rs

```diff
@@ -70,12 +70,17 @@ impl TcpStream {
         Self { inner }
     }
 
-    /// Read some data from the stream into the buffer, returning the original buffer and
-    /// quantity of data read.
+    /// Read some data from the stream into the buffer.
+    ///
+    /// Takes ownership of the buffer for the duration of the Future.
```

Comment:
I'm not sure about this. Again, if this method has a comment that calls out this detail about an argument, are all the other methods that do the same now outdated and incorrect in their admission? But I'll leave it to you and Noah to work out.

Reply:
Ok.. Removed all references to functions taking ownership.